### ISSUE TITLE
Add support for authenticated media requests using access token

### ISSUE THREAD

**Issue by girlbossceo** [girlbossceo/conduwuit#105]
Helps prevent people from proxying unwanted media from/to your homeserver as they require authentication using a valid access token on your server. conduwuit is fast so it makes sense why those people would choose these kinds of homeservers to do this with.

I think it was `neko.dev` that had this example in production.

Blocked on my findings with https://github.com/girlbossceo/conduwuit/issues/18#issuecomment-1900159920 as we need proper infrastructure for forcing endpoints to require a valid access token.

**Comment by girlbossceo** [comment]
An official authenticated media MSC was merged into spec, but ended up basically redoing the entirety of media.
